src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ComponentResourceManager.cs

```diff
@@ -149,10 +149,10 @@ public virtual void ApplyResources(object value, string objectName, CultureInfo
                     }
                 }
 
-                // Character after objectName.Length should be a ".", or else we should continue.
+                // Character after objectName.Length should be a "." or a '-', or else we should continue.
```

Comment:
Correct. I just talked to @danmosemsft and explained offline the possible case that could hit :). 

Thanks for putting it here @tarekgh